File: python/167_Two-Sum-II-Input-Array-Is-Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if len(numbers)==2: return [1, 2]

        ptr1, ptr2 = 0, len(numbers)-1

        while(ptr2>ptr1):
            if (numbers[ptr1]+numbers[ptr2]) > target:
                ptr2-=1
            elif (numbers[ptr1]+numbers[ptr2]) < target:
                ptr1+=1
            else:
                return [ptr1+1, ptr2+1]

File: c++/167_Two-Sum-II-Input-Array-Is-Sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int low=0, high=numbers.size()-1;

        while(low<high){
            int twoSum = numbers[low]+numbers[high];
            if(twoSum==target) return vector<int>{low+1, high+1};
            else if (twoSum > target) --high;
            else ++low;
        }
        //no valid solution
        return vector<int>{-1};
    }
};